Infrastructure/DataImporterService.cs

```diff
@@ -104,26 +104,41 @@ private async Task ImportTutionPartnerLogos(NtpDbContext dbContext, ILogoFileEnu
 
         _logger.LogInformation("Looking for logos for {Count} tuition partners", partners.Count);
 
-        var logos = logoFileEnumerable.ToList();
-        _logger.LogInformation("Available logo files are {LogoFiles}",
-            string.Join("\n", logos.Select(x => x.Filename).OrderBy(x => x)));
-
-        foreach (var partner in partners)
+        var logos = logoFileEnumerable.Where(x => x.Filename.EndsWith(".svg")).ToList();
+
+        var matching = (from p in partners
+                        from l in logos
+                        where IsFileLogoForTuitionPartner(p.SeoUrl, l.Filename)
+                        select new
+                        {
+                            Partner = p,
+                            Logo = l,
+                        })
+                       .ToList();
+
+        _logger.LogInformation("Matched {Count} logos to tuition partners:\n{Matches}",
+            matching.Count, string.Join("\n", matching.Select(x => $"{x.Partner.SeoUrl} => {x.Logo.Filename}")));
+
+        var partnersWithoutLogos = partners.Except(matching.Select(x => x.Partner));
+        if (partnersWithoutLogos.Any())
         {
-            var dataFile = logos
-                .Where(logo => logo.Filename.Contains(partner.SeoUrl))
-                .FirstOrDefault();
+            _logger.LogInformation("{Count} tuition partners do not have logos:\n{WithoutLogo}",
+                partnersWithoutLogos.Count(), string.Join("\n", partnersWithoutLogos.Select(x => x.SeoUrl)));
+        }
 
-            if (dataFile == null)
-            {
-                _logger.LogInformation("No logo file for Tution Partner {Name}", partner.SeoUrl);
-                continue;
-            }
+        var logosWithoutPartners = logos.Except(matching.Select(x => x.Logo));
```

Comment:
Worth adding a .ToList or array here and for logosWithoutPartners to remove the possible multiple enumerations warning?

Reply:
It's only enumerated twice so it's hard to know whether the allocations to turn it into a list/array would be better or worse than running through the up to 54 (currently 21) logos twice.

I don't see that warning - is that resharper?